Meta/gn/build/libs/simdutf/BUILD.gn

```diff
@@ -1,6 +1,10 @@
 import("//Meta/gn/build/libs/third_party.gni")
 
 third_party_dependency("simdutf") {
-  libs = [ "simdutf" ]
+  if (current_os == "win") {
+    libs = [ "simdutf.lib" ]
```

Comment:
Wonder if this could be done at a lower level - in `third_party_dependency` or in whatever is consuming the `libs` variable. Would be nice not to need a Windows branch on every definition of `libs`.

Reply:
Yeah, I was thinking of sinking this into `third_party_dependency` but wasn't sure how to rewrite the `libs` parameter. However, its also unclear at that level the library type (i.e. static vs dynamic and thus needing the import lirbary).

Reply:
I believe that whatever consumes the `libs` variable is going to be the "solink", "solink_module" and "link" `tool` rules in Meta/gn/build/toolchain/BUILD.gn. Not sure what the recommended way to handle import libs for gn projects is though. feels like a @nico question :grimacing:

Reply:
In theory we would want a "debug" build to use shlibs for faster link times, and a "release" build to use static libs? Though Tim's latest PR for vcpkg was to enable shlibs for the default RelWithDebInfo developer build anyway so ... :shrug: . We might end up needing special cases everywhere regardless based on config. it would be nice if we could treat every lib the same, but I'm not sure if that's 100% possible (we might end up linking against a windows-shipped zlib static lib or something for example).

Reply:
> In theory we would want a "debug" build to use shlibs for faster link times, and a "release" build to use static libs?

In practice, we will also want release builds with shared linkage - people are going to be building optimized builds, and will want shared linking as well.

So instead of picking shared vs. static based on the build type, I think we'll want it as a separate option. Chromium does this with a `is_component_build` args.gn option.

Reply:
I think that there is more complexity than that. I know that for distributing the Swift toolchain, I end up having to test various combinations to determine what makes better sense - static or dynamic linking. I don't think that we can do a wholesale dynamic or static, and would want either more knobs or some sort of pre-defined set.

Reply:
> I believe that whatever consumes the `libs` variable is going to be the "solink", "solink_module" and "link" `tool` rules in Meta/gn/build/toolchain/BUILD.gn. Not sure what the recommended way to handle import libs for gn projects is though. feels like a @nico question 😬

You could do something like

```
    if (current_os == "win") {
      new_libs = []
      foreach(lib, libs) {
        new_libs += [ lib + ".lib" ]
      }
      libs = new_libs
    }
```

in third_party.gni (untested, but roughly the right direction I think).

(Building against non-system-provided libs isn't super common in GN land, but it should work.)

Reply:
Thanks @nico - that is definitely a good approach so that we don't need to have the knowledge of Windows specifics be a requirement for everyone.